### PR TITLE
Seal ArrowPrimitiveType

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

As ArrowNativeType is sealed, it is hard to see how or why anyone would implement this for a custom type, but sealing it makes this contract very clear.

# What changes are included in this PR?

# Are there any user-facing changes?

Technically this is a breaking change, practically I sincerely doubt anyone will notice

### CODE REVIEW

File: arrow-array/src/array/primitive_array.rs [hunk 1]
```diff
@@ -230,27 +230,7 @@ pub type Decimal128Array = PrimitiveArray<Decimal128Type>;
 /// scale less or equal to 76.
 pub type Decimal256Array = PrimitiveArray<Decimal256Type>;
 
-/// Trait bridging the dynamic-typed nature of Arrow (via [`DataType`]) with the
-/// static-typed nature of rust types ([`ArrowNativeType`]) for all types that implement [`ArrowNativeType`].
-pub trait ArrowPrimitiveType: 'static {
```
Comment (tustvold) [inline arrow-array/src/array/primitive_array.rs:235]: I moved this into types along with all the other traits of this sort

File: arrow-array/src/types.rs [hunk 2]
```diff
@@ -240,24 +267,12 @@ impl ArrowDictionaryKeyType for UInt32Type {}
 
 impl ArrowDictionaryKeyType for UInt64Type {}
 
-mod run {
-    use super::*;
-
-    pub trait RunEndTypeSealed {}
```
Comment (tustvold) [inline arrow-array/src/types.rs:246]: This is no longer necessary, as PrimitiveType is sealed

**Review by alamb — APPROVED** [review]
LGTM -- @jhorstmann  and @viirya  may be interested

**Comment by jhorstmann** [comment]
Fully agree with making this sealed :+1:

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Making sense. 👍

**Pull request merged**